### PR TITLE
Add `TargetInfo` wrapper for `sgx_target_info_t`

### PR DESCRIPTION
**Pull request by nick-mobilecoin**

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #91** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/91" target="_blank">Graphite</a> 
    * **PR #86** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/86" target="_blank">Graphite</a> 
      * **PR #92** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/92" target="_blank">Graphite</a> 
        * **PR #103** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/103" target="_blank">Graphite</a> 
          * **PR #110** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/110" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/110?utm_source=stack-comment).

File: core/types/src/svn.rs [hunk 1]
```diff
@@ -39,3 +42,25 @@ pub struct CpuSvn(sgx_cpu_svn_t);
 impl_newtype_for_bytestruct! {
     CpuSvn, sgx_cpu_svn_t, SGX_CPUSVN_SIZE, svn;
 }
+
+/// Config ID
```
Comment (nick-mobilecoin) [inline core/types/src/svn.rs:46]: Not fond of the config id landing in `svn.rs`.  Looking at the sgx c code it looks like it's in `sgx_key.h`, it seems that it's a type without a clear home.

**Review by jcape — COMMENTED** [review]
LGTM save for `ConfigId`---in attest-core I put it in it's own module (it shows up in, like, the XML generation for `sgx_sign`, but it can live wherever).

File: core/types/src/report.rs [hunk 2]
```diff
@@ -2,14 +2,13 @@
 //! SGX Report
 
 use crate::{
-    impl_newtype_for_bytestruct, new_type_accessors_impls, Attributes, ConfigSvn, CpuSvn, IsvSvn,
-    Measurement, MiscellaneousSelect, MrEnclave, MrSigner,
+    impl_newtype_for_bytestruct, new_type_accessors_impls, svn::ConfigId, Attributes, ConfigSvn,
```
[the remark this answers is not in the document]
Comment (nick-mobilecoin) [reply core/types/src/report.rs]: Exported ConfigId at top level and moved to a separate module

**Review by awygle — APPROVED** [review]
Looks good!

File: core/types/src/target_info.rs [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+            attributes: Attributes::default().into(),
+            reserved1: [0u8; SGX_TARGET_INFO_RESERVED1_BYTES],
+            config_svn: ConfigSvn::default().into(),
+            misc_select: MiscellaneousSelect::default().into(),
+            reserved2: [0u8; SGX_TARGET_INFO_RESERVED2_BYTES],
+            config_id: ConfigId::default().into(),
+            reserved3: [0u8; SGX_TARGET_INFO_RESERVED3_BYTES],
+        })
+    }
+}
```
Comment (jcape) [inline core/types/src/target_info.rs:62]: This should probably get deleted eventually (though `sgx_target_info_t` should keep it) in favor of a monkey-patched API in `mc-sgx-urts`:

```rust
pub trait TryFromId {
    fn try_from_id(id: EnclaveId) -> Self;
}

impl TryFromId for TargetInfo {
    fn try_from_id( id: EnclaveId) -> Result<Self, SgxError> {
        let mut info = sgx_target_info_t::default();
        unsafe{ sgx_get_target_info(id, &mut info as *mut _) }
            .into_result()?
            .map(|_| TargetInfo::from(info))
}
```

**Review by jcape — APPROVED** [review]
LGTM, though we should remove `TargetInfo::default()` at some point

File: core/types/src/target_info.rs (hunk 3 above)
Comment (nick-mobilecoin) [reply core/types/src/target_info.rs:62]: the Default for `TargetInfo` goes away in #115

**Comment by nick-mobilecoin** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/110?utm_source=stack-comment).


Job Reference: C6ZyZ0Ggec8V0ksW3mYN

**Pull request merged**